src/main/scala/firrtl/backends/proto/ProtoBufEmitter.scala

```diff
@@ -0,0 +1,115 @@
+// SPDX-License-Identifier: Apache-2.0
+package firrtl.backends.proto
+
+import firrtl.{AnnotationSeq, CircuitState, DependencyAPIMigration, Transform}
+import firrtl.ir
+import firrtl.annotations.NoTargetAnnotation
+import firrtl.options.CustomFileEmission
+import firrtl.options.Viewer.view
+import firrtl.proto.ToProto
+import firrtl.stage.{FirrtlOptions, Forms}
+import firrtl.stage.TransformManager.TransformDependency
+import java.io.{ByteArrayOutputStream, Writer}
+
+/** This object defines Annotations that are used by Protocol Buffer emission.
+  */
+object Annotation {
+
+  /** This will cause the enclosed circuit to be serialized to a Protocol Buffer with the provided suffix. */
+  case class ProtoBufSerialization(circuit: ir.Circuit, suffix: Option[String])
+      extends NoTargetAnnotation
+      with CustomFileEmission {
+
+    /** The hash code is cached for this case class because it includes a very large circuit operand whose hash is expensive
+      * to compute.  This is safe to do because the case class is immutable.
+      */
+    override lazy val hashCode = scala.util.hashing.MurmurHash3.productHash(this)
+
+    override protected def baseFileName(annotations: AnnotationSeq): String = {
+      view[FirrtlOptions](annotations).outputFileName.getOrElse(circuit.main)
+    }
+
+    override def getBytes: Iterable[Byte] = {
+      val ostream = new java.io.ByteArrayOutputStream
+      ToProto.writeToStream(ostream, circuit)
+      ostream.toByteArray()
+    }
+  }
+
+}
+
+/** Running this transform will cause a circuit to be emitted to ProtoBuf after some prerequisites have been
+  * satisfied.
+  *
+  * This is not intended to be used directly.  Instead see one of the concrete emitters, e.g., [[Emitter.Low]].
+  * @see [[Emitter.Chirrtl]]
+  * @see [[Emitter.MHigh]]
+  * @see [[Emitter.High]]
+  * @see [[Emitter.Middle]]
+  * @see [[Emitter.Low]]
+  * @see [[Emitter.OptLow]]
+  */
+sealed abstract class ProtoBufEmitter(prereqs: Seq[TransformDependency])
+    extends Transform
+    with DependencyAPIMigration
+    with firrtl.Emitter {
+
+  override def prerequisites = prereqs
+  override def optionalPrerequisites = Seq.empty
+  override def optionalPrerequisiteOf = Seq.empty
+  override def invalidates(a: Transform) = false
+
+  override def execute(state: CircuitState) =
+    state.copy(annotations = state.annotations :+ Annotation.ProtoBufSerialization(state.circuit, Some(outputSuffix)))
+
+  override def emit(state: CircuitState, writer: Writer) = ???
```

Comment:
This probably should give a more specific error, eg. 
```suggestion
  override def emit(state: CircuitState, writer: Writer) = {
    val msg = "This emitter works differently than the others and is intended to be used only as a transform"
    throw new scala.NotImplementedError(msg)
  }
```

Reply:
Or perhaps this class shouldn't mixin `firrtl.Emitter`? Is it gaining anything for doing so?

Reply:
Will just implement this. It's annoying tedious to decouple this from extending `Emitter`. (There are a bunch of places that expect an `Emitter` to be around.) Future work...